backend/app/api/endpoints/mcp_providers.py

```
@@ -0,0 +1,134 @@
+) -> MCPProviderListResponse:
+    """List all available MCP providers"""
+    # Parse preferences from JSON string
+    parsed_prefs = _parse_user_preferences(current_user.preferences)
+    providers = MCPProviderService.list_providers(parsed_prefs)
+    return MCPProviderListResponse(providers=providers)
+
+
+@router.post("/{provider_key}/servers", response_model=MCPServerListResponse)
+async def get_provider_servers(
+    provider_key: str,
+    db: Session = Depends(get_db),
```

Ruff (0.14.14)

[warning] Unused function argument: `db`

(ARG001)